devtools: document and further automate release process

dnlggr opened the pull request:
Resolves #105.

Building on the feedback in #103, this integrates `npm version` into the release process.

Now, when you run e.g. `npm version patch`, it will:
1. Bump the version
2. Update the changelog
3. Commit the changes to a branch
4. Push the branch and open a _draft_ PR. (Let me know if we should add a manual step in between here.)

See the new RELEASING.md for more details.

This is just a suggestion. My main goal is to make everything as painless and automated as possible while sticking close to npm tooling. Please let me know what you think of it and what we could do better!

theborakompanioni reviewed (approved):
I think for the first few releases this is perfectly sufficient. Good work, thanks! :rocket: 

I did not test the "postversion" command.

dnlggr commented:
Fixed two things:

- Updated the docs to mention that a PR can only be crated if you have the GitHub CLI installed.
- Removed the review request from the PR since it'll be pushed as a draft.
